utils/__tests__/messages.test.ts

```diff
@@ -0,0 +1,164 @@
+  departmentName: "MyDeptName" as NonEmptyString,
+  isVisible: true,
+  maxAllowedPaymentAmount: 0 as MaxAllowedPaymentAmount,
+  organizationFiscalCode: anOrganizationFiscalCode,
+  organizationName: "MyOrgName" as NonEmptyString,
+  requireSecureChannels: false,
+  serviceId: "MySubscriptionId" as NonEmptyString,
+  serviceName: "MyServiceName" as NonEmptyString
+};
+
+const aNewService: NewService = {
+  ...aService,
+  kind: "INewService"
+};
+
+const aRetrievedService: RetrievedService = {
+  ...aNewService,
+  ...aCosmosResourceMetadata,
+  id: "123" as NonEmptyString,
+  kind: "IRetrievedService",
+  version: 1 as NonNegativeInteger
+};
+
+const aFiscalCode = "FRLFRC74E04B157I" as FiscalCode;
+const aDate = new Date();
+
+const aNewMessageWithoutContent: NewMessageWithoutContent = {
+  createdAt: aDate,
+  fiscalCode: aFiscalCode,
+  id: "A_MESSAGE_ID" as NonEmptyString,
+  indexedId: "A_MESSAGE_ID" as NonEmptyString,
+  isPending: true,
+  kind: "INewMessageWithoutContent",
+  senderServiceId: "test" as ServiceId,
+  senderUserId: "u123" as NonEmptyString,
+  timeToLiveSeconds: 3600 as TimeToLiveSeconds
+};
+
+const aRetrievedMessageWithoutContent: RetrievedMessageWithoutContent = {
+  ...aNewMessageWithoutContent,
+  ...aCosmosResourceMetadata,
+  kind: "IRetrievedMessageWithoutContent"
+};
+
+const blobServiceMock = ({
+  getBlobToText: jest.fn()
+} as unknown) as BlobService;
+
+const messageModelMock = ({
+  getContentFromBlob: () =>
+    TE.of(
+      O.some({
+        subject: "a subject",
+        markdown: "a markdown"
+      } as MessageContent)
+    )
+} as unknown) as MessageModel;
+
+const serviceModelMock = ({
+  findLastVersionByModelId: () => TE.of(O.some(aRetrievedService))
+} as unknown) as ServiceModel;
+
+describe("Messages", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("should return right when message blob and service are correctly retrieved", async () => {
+    const messages = [
+      retrievedMessageToPublic(aRetrievedMessageWithoutContent)
+    ] as readonly CreatedMessageWithoutContent[];
+
+    const enrichMessages = enrichMessagesData(
+      messageModelMock,
+      serviceModelMock,
+      blobServiceMock
+    );
+
+    const enrichedMessagesPromises = enrichMessages(messages);
+
+    const enrichedMessages = await pipe(
+      TE.tryCatch(async () => Promise.all(enrichedMessagesPromises), void 0),
+      TE.getOrElse(() => {
+        throw Error();
+      })
+    )();
+
+    enrichedMessages.map(enrichedMessage => {
+      expect(E.isRight(enrichedMessage)).toBe(true);
+      if (E.isRight(enrichedMessage)) {
+        expect(EnrichedMessage.is(enrichedMessage.right)).toBe(true);
+      }
+    });
+  });
+
+  it("should return left when message blob or service are not correctly retrieved", async () => {
+    serviceModelMock.findLastVersionByModelId = jest
+      .fn()
+      .mockImplementationOnce(() => TE.left(E.toError("error")));
```

Comment:
FYI

`E.toError` is an utility to convert different kind of values into `Error` instances. In this case, `new Error("error")` is the very same.

Reply:
Fixe in https://github.com/pagopa/io-functions-app/pull/203/commits/1cf1b28e1b0a804d1843054480a4029665ad1e53